### PR TITLE
Fix vso commands execution in vulnerable variables

### PR DESCRIPTION
**Pull request by KonstantinTyukalov**
Description:
`*.DefinitionName` and `*.SourceVersionMessage` may contain vso commands which will be executed if it will be specified in script as output
We want to prevent this behavior by deactivating commands in this variables

Changelog:
- Added `DeactivateVsoCommands` util which disables all vso commands in line
- Added Scrubbing of vso commands from job message variables which may be echoed in customer's pipelines
- Added unit tests

### CODE REVIEW

File: src/Agent.Worker/Variables.cs [hunk 1]
```diff
@@ -260,6 +260,14 @@ public bool Retain_Default_Encoding
             "RequestedFor"
         };
 
+        public static readonly List<string> VariablesVulnerableToExecution = new List<string>()
```
Comment (Roman-Shchukin) [inline src/Agent.Worker/Variables.cs:263]: Parentheses at the end could be removed

File: src/Test/L0/Worker/WorkerL0.cs [hunk 2]
```diff
@@ -228,6 +228,64 @@ public void VerifyJobRequestMessagePiiDataIsScrubbed()
...
+        [Trait("Category", "Worker")]
+        public void VerifyJobRequestMessageVsoCommandsScrubbed()
+        {
+            Pipelines.AgentJobRequestMessage message = CreateJobRequestMessage("jobwitvsocommands");
+
+            message.Variables[Constants.Variables.Build.SourceVersionMessage] = "##vso[setVariable]etc1";
+            message.Variables[Constants.Variables.System.SourceVersionMessage] = "##vso[setVariable]etc2";
+            message.Variables[Constants.Variables.Build.DefinitionName] = "##vso[setVariable]etc3";
+            message.Variables[Constants.Variables.System.DefinitionName] = "##vso[setVariable]etc4";
+
```
Comment (Roman-Shchukin) [inline src/Test/L0/Worker/WorkerL0.cs:242]: Extra empty line

File: src/Agent.Worker/Variables.cs (hunk 1 above)
Comment (KonstantinTyukalov) [reply src/Agent.Worker/Variables.cs]: Fixed, thanks

File: src/Test/L0/Worker/WorkerL0.cs (hunk 2 above)
Comment (KonstantinTyukalov) [reply src/Test/L0/Worker/WorkerL0.cs:242]: Thanks, removed not needed variable

**Review by Roman-Shchukin — APPROVED** [review]

**Review by vmapetr — APPROVED** [review]

**Review by kirill-ivlev — APPROVED** [review]

**Review by max-zaytsev — APPROVED** [review]

**Review by wojgasior — APPROVED** [review]

**Comment by KonstantinTyukalov** [comment]
not added trace events to debug potential problems, will do

**Pull request merged**